source/extensions/filters/network/thrift_proxy/router/router_impl.cc

```diff
@@ -503,6 +508,9 @@ void Router::UpstreamRequest::onPoolReady(Tcp::ConnectionPool::ConnectionDataPtr
     upgrade_response_ =
         protocol_->attemptUpgrade(*transport_, *conn_state_, parent_.upstream_request_buffer_);
     if (upgrade_response_ != nullptr) {
+      parent_.recordClusterScopeHistogram({parent_.upstream_rq_size_},
```

Comment:
good point to record it here too - if we can cover this in a test too that would be great

Reply:
Done.